### DISCUSSION TITLE
Upgrade pc without internet

### DISCUSSION THREAD

**Discussion by Muncubus** [AveYo/MediaCreationTool.bat#80] [Q&A] [answered]
Good day.
I want to upgrade pc's with  Windows 7 without internet connection.
What I need to do step by step to create an upgrade package, which I can copy to to machine and run script? 
Thank a lot.

**Comment by AveYo** [comment, marked as answer]
NOPE.
Run the script on a pc ***with*** internet connection and create a setup media, 
preferably on a USB stick, but ISO is fine too if you burn it to a DVD later on.
Use that setup media (USB stick or DVD) on the _pc's with Windows 7 without internet connection_.
Click auto.cmd instead of setup.exe to facilitate automatic installation keeping files and apps

_Note that direct upgrade from 7 to 11 does not allow keeping files and apps._

**Reply by Muncubus** [reply]
I don't need 11...
I ran the script with parameters set MCT=21H2 set /a ISO=1 and got several folders:
C:\$Windows.~WS 
C:\ESD 
As I understand in ESD folder is original image and in C:\$Windows.~WS\Sources\Windows\ with auto.cmd is what I need?
I just copy windows folder to any folder on destination pc and run cmd file, right?

**Comment by AveYo** [comment]
The script creates the iso in the same path (if it's writable) else in C:\ESD as a fallback.
That's what you need. Then you mount that iso or burn it to a DVD or usb stick.

Why don't you create a USB stick, it would make everything easier since you don't need to copy anything extra, just plug the stick in the other PCs?

And if you make an iso, it's easier and less error-prone to just rename the script than edit it
i.e. `21H2 iso MediaCreationTool.bat` and run it from a writable directory such as Downloads or Documents

If you're doing all this on windows 7, it does not know natively to mount iso files, you need 3rd party software for that (example: [Virtual CloneDrive](https://www.redfox.bz/en/virtual-clonedrive.html))

**Reply by Muncubus** [reply]
No iso file anywhere, but actually I don't need it. Copying this folder (C:$Windows.~WS\Sources\Windows\) to win7 pc and running auto.cmd solves problem. Thanks a lot.
